src/nwb_benchmarks/_flask/_app.py

```diff
@@ -0,0 +1,422 @@
+            result = manager.ensure_repo_up_to_date()
+            if result is not None:
+                return result
+
+            time.sleep(1)
+
+            repackage_as_parquet(directory=directory, output_directory=output_directory)
+
+            time.sleep(1)
+
+            result = manager.add_and_commit(message="Update benchmark databases")
+            if result is not None:
+                return result
+
+            time.sleep(1)
+
+            manager.push()
+
+            return 200
+        except Exception as exception:
+            return {"type": type(exception).__name__, "error": str(exception), "traceback": traceback.format_exc()}, 500
+
+
+class GitHubResultsManager:
+    def __init__(self, repo_name: str):
+        self.cache_directory = Path.home() / ".cache" / "nwb-benchmarks"
+        self.cache_directory.mkdir(parents=True, exist_ok=True)
+        self.repo_path = self.cache_directory / repo_name
+
+    def ensure_repo_up_to_date(self) -> typing.Literal[521, 522] | None:
+        """Clone repository if it doesn't exist locally."""
+        if not self.repo_path.exists():
+            return 521
+
+        command = f"git pull"
+        cwd = self.repo_path
+        result = subprocess.run(
+            args=command,
+            cwd=cwd,
+            capture_output=True,
+            shell=True,
+        )
+        if result.returncode != 0:
+            message = f"Git command ({command}) failed: {result.stderr.decode()}"
+            print(f"ERROR: {message}")
+            return 522
+
+    def write_file(self, filename: str, json_content: dict) -> None:
+        """Write results JSON to a file in the cache directory."""
+        base_directory = self.cache_directory / "nwb-benchmarks-results"
+        if filename.startswith("environment-"):
+            directory = base_directory / "environments"
+        if filename.startswith("machine-"):
+            directory = base_directory / "machines"
+        elif filename.endswith("_results"):
+            directory = base_directory / "results"
+        else:
+            # Legacy outer collection
+            directory = base_directory
+        file_path = directory / filename
+
+        with open(file=file_path, mode="w") as file_stream:
+            json.dump(obj=json_content, fp=file_stream, indent=4)
+
+    def add_and_commit(self, message: str) -> typing.Literal[523] | None:
+        """Commit results to git repo."""
+        command = f"git add . && git commit -m '{message}'"
+        result = subprocess.run(
+            args=command,
+            cwd=self.repo_path,
+            capture_output=True,
+            shell=True,
+        )
+        if result.returncode != 0:
+            message = f"Git command ({command}) failed: {result.stderr.decode()}\ntraceback: {traceback.format_exc()}"
+            print(f"ERROR: {message}")
+            raise RuntimeError(message)
+            # return 523
```

Comment:
Would a RuntimeError crash the app? Should this be `return 523` again?

Reply:
Anyway, reverted to the previous 523 now